File: src/main/java/io/github/dylmeadows/eontimer/model/config/Console.java
package io.github.dylmeadows.eontimer.model.config;

import io.github.dylmeadows.common.javafx.util.Choice;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor(access = AccessLevel.PACKAGE)
public enum Console implements Choice {
    GBA("GBA", 59.7271),
    NDS("NDS", 59.8261),
    DSI("DSI", 59.8261),
    _3DS("3DS", 59.8261);

    private final String text;
    private final double fps;

    public double getFrameRate() {
        return 1000 / fps;
    }
}

File: src/main/kotlin/io/github/dylmeadows/eontimer/model/settings/Console.kt
package io.github.dylmeadows.eontimer.model.settings

import io.github.dylmeadows.common.javafx.util.Choice

enum class Console(private val text: String, val fps: Double) : Choice {
    GBA("GBA", 59.7271),
    NDS("NDS", 59.8261),
    DSI("DSI", 59.8261),
    _3DS("3DS", 59.8261);

    val frameRate: Double get() = 1000 / fps

    override fun getText(): String {
        return this.text
    }
}